extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-preview.js

```diff
@@ -0,0 +1,735 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpStoryPlayer} from '../../../src/amp-story-player/amp-story-player-impl';
+import {
+  addAttributeAfterTimeout,
+  removeAfterTimeout,
+  updateHash,
+} from './utils';
+import {closest} from '../../../src/dom';
+import {escapeCssSelectorIdent} from '../../../src/css';
+import {htmlFor} from '../../../src/static-template';
+import {setStyles} from '../../../src/style';
+
+/**
+ * Creates a tab content, will be deleted when the tabs get implemented.
+ * @param {!Window} win
+ * @param {string} storyUrl
+ * @param {?string} devices
+ * @return {!Element} the layout
+ */
+export function createTabPreviewElement(win, storyUrl, devices) {
+  const element = win.document.createElement('amp-story-dev-tools-tab-preview');
+  element.setAttribute('data-story-url', storyUrl);
+  devices ? element.setAttribute('data-devices', devices) : null;
+  return element;
+}
+
+/**
+ * Generates the template for a device.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildDeviceTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-dev-tools-device">
+      <div class="i-amphtml-story-dev-tools-device-screen">
+        <amp-story-player width="1" height="1" layout="container">
+          <a></a>
+        </amp-story-player>
+      </div>
+    </div>
+  `;
+};
+
+/**
+ * Generates the template for a device chip.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildDeviceChipTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <span
+      class="i-amphtml-story-dev-tools-device-chip"
+      data-action="toggleDeviceChip"
+    >
```

Comment:
A11y:
These should be button elements.

Doing this can potentially help every user. Tabbing through a page is a fast way to navigate.

Reply:
Good call, working on it. Many of the other interactive elements are also buttons, like the tab names, "open story" button, etc.